modules/jax/src/main/ts/ephox/jax/core/HttpJwt.ts

```diff
@@ -10,24 +10,25 @@ const headers = (headersInput: HttpRequest<ResponseBodyDataTypes>['headers'], to
   return headersInput ? { ...headersInput, ...authHeader } : authHeader;
 };
 
-type RunMethod<T> = (token: JwtToken) => FutureResult<T, HttpError>;
+type RunMethod<T, U extends ResponseType> = (token: JwtToken) => FutureResult<T, HttpError<U>>;
```

Comment:
Maybe use `E` instead of `U` here and below as this is capturing the "Error" response type?

Reply:
Also do we want to allow different response types here between a success and failure? We don't allow it in `put`, `post`, etc... In which case this might be better as something like this:
```suggestion
type RunMethod<T extends ResponseType> = (token: JwtToken) => FutureResult<ResponseTypeMap[T], HttpError<T>>;
```
and then we can update the places below to just pass `T extends ResponseType`.

Reply:
Fixed.